### PR TITLE
refactor(lint/noConfusingArrow): remove rule

### PR DESCRIPTION
**Pull request by Conaclos**
## Summary

Remove `noConfusingArrow` which is mostly useless.
Because it is a nursery rule, it is not a breaking change.

## Test Plan

Removed tests.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -32,6 +32,8 @@ Read our [guidelines for writing a good changelog entry](https://github.com/biom
 
 - [useCollapsedElseIf](https://biomejs.dev/linter/rules/use-collapsed-else-if/) now only provides safe code fixes. Contributed by [@Conaclos](https://github.com/Conaclos)
 
+- Remove [noConfusingArrow](https://biomejs.dev/linter/rules/no-confusing-arrow/).
```
Comment (ematipico) [inline CHANGELOG.md:35]: Maybe it's worth saying why we removed, just a small sentence

**Review by ematipico — APPROVED** [review]

**Comment by innocenzi** [comment]
Some people [use eslint instead of prettier](https://antfu.me/posts/why-not-prettier), because we don't agree with all of prettier's formatting choices and it's not really configurable.

I'm investing Biome as my eslint replacement for when it will support Vue, and I saw that removal in the changelog.

My question is, how viable would it be to keep using the same "strategy" (no formatter if not configurable, only linter with formatting rules) with Biome?

**Comment by Conaclos** [comment]
> My question is, how viable would it be to keep using the same "strategy" (no formatter if not configurable, only linter with formatting rules) with Biome?

I am not sure to understand your question.
Since the start, Biome (and Rome before), avoids implementing formatting-related rules. This is mainly to avoid a maintenance burden. The linter can be used without the formatter, and vice versa. We are open to add some configuration on the formatter when there is a widespread convention. You can open a discussion to discuss a particular point. I personally could like to have a way to disable line hard-wrapping in the future.

**Comment by innocenzi** [comment]
Thanks, I think that answers my question 👍

**Pull request merged**